src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,58 @@
+package duke.task;
+
+import java.time.LocalDate;
+
+public class Task {
+    protected String description;
+    protected boolean isDone;
+
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    public Task(String description, Boolean isDone) {
+        this.description = description;
+        this.isDone = isDone;
+    }
+
+    public String getStatusIcon() {
+        return (isDone ? "[X]" : "[ ]"); // mark done task with X
+    }
+
+    public boolean getStatus() {
+        return isDone;
+    }
+
+    public String getDescription() {
+        return this.description;
+    }
+
+    public void markAsDone() {
+        this.isDone = true;
+    }
+
+    public void markAsUndone() {
+        this.isDone = false;
+    }
+
+    public String taskType() {
```

Comment:
This is a method that does a certain action, so it might be better to name it as "getTaskType"?